File: competitive_programming/leetcode/06_javascript/7_reverse_integer.js
function reverseInteger(x) {
  const sign = x < 0 ? -1 : 1;
  const reversed = sign * Number(String(Math.abs(x)).split("").reverse().join(""));
  return reversed >= -(2 ** 31) && reversed <= 2 ** 31 - 1 ? reversed : 0;
}

if (require.main === module) {
  if (reverseInteger(123) !== 321 || reverseInteger(-123) !== -321 || reverseInteger(120) !== 21) throw new Error("reverse integer failed");
  if (reverseInteger(1534236469) !== 0) throw new Error("reverse integer overflow failed");
  console.log("7 reverse integer ok");
}

File: competitive_programming/leetcode/11_golang/7_reverse_integer.go
package main

import "fmt"

func reverseInteger(x int) int {
	sign := 1
	if x < 0 {
		sign = -1
	}
	value := x * sign
	reversed := 0
	for value > 0 {
		reversed = reversed*10 + value%10
		value /= 10
	}
	reversed *= sign
	if reversed < -(1<<31) || reversed > (1<<31)-1 {
		return 0
	}
	return reversed
}

func main() {
	if reverseInteger(123) != 321 || reverseInteger(-123) != -321 || reverseInteger(120) != 21 {
		panic("reverse integer failed")
	}
	if reverseInteger(1534236469) != 0 {
		panic("reverse integer overflow failed")
	}
	fmt.Println("7 reverse integer ok")
}
